adaptive_components/lib/src/adaptive_container.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:adaptive_breakpoints/adaptive_breakpoints.dart';
+import 'package:flutter/material.dart';
+
+/// [AdaptiveContainer] lets you create a [Container] with adaptive constraints.
+///
+/// The AdaptiveContainer does everything a normal container does but with
+/// adaptive constraints. For more information go to one of the links below.
+///
+/// https://api.flutter.dev/flutter/widgets/Container-class.html
+/// https://material.io/design/layout/responsive-layout-grid.html#breakpoints
+///
+/// This class is useful whenever you want a container to only be active in
+/// certain [AdaptiveWindowType].
+class AdaptiveContainer extends StatelessWidget {
+  /// Creates a widget that combines common painting, positioning, and sizing widgets.
+  ///
+  /// The `color` and `decoration` arguments cannot both be supplied, since
+  /// it would potentially result in the decoration drawing over the background
+  /// color. To supply a decoration with a color, use `decoration:
+  /// BoxDecoration(color: color)`.
+  AdaptiveContainer({
+    Key key,
+    this.alignment,
+    this.padding,
+    this.color,
+    this.decoration,
+    this.foregroundDecoration,
+    this.margin,
+    this.transform,
+    this.height,
+    this.child,
+    this.clipBehavior = Clip.none,
+    this.adaptiveConstraints,
+    this.adaptiveColumn = 1,
```

Comment:
is this columnWidth?

Reply:
No, it represents the number of columns it should consume. So if is 4 and there are 3 adaptvie columns in a xlarge screen than it the adaptive columns would take up the row. But if it was a small screen that the adaptive columns would appear as a regular column

Reply:
Maybe a better name would be columns or adaptiveColumns (although the adaptive prefix is not really necessary since we are in context of an AdaptiveContainer)